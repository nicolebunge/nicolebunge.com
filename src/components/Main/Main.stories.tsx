import Main from '.';

const meta = {
  component: Main,
};

const Default = {
  args: {
    children: 'Main',
  },
};

export { Default };
export default meta;
