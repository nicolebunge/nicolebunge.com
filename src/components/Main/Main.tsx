import clsx from 'clsx';
import React, { HTMLAttributes, ReactNode } from 'react';
import { main } from './Main.module.css';

interface MainProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
}

function Main(props: MainProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <main className={clsx(className, main)} role="main" {...otherProps} />;
}

export type { MainProps };
export default Main;
