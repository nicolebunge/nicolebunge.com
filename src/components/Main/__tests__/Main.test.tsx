import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../Main.stories';

const { Default } = composeStories(stories);

describe('Main component', () => {
  it('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
