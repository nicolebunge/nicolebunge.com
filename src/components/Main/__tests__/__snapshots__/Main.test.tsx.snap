// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main component > renders correctly 1`] = `
<main
  class="main"
  role="main"
>
  Main
</main>
`;
